lib/karafka/base_controller.rb

```diff
@@ -98,19 +34,27 @@ def params_batch=(messages)
     end
 
     # Executes the default controller flow, runs callbacks and if not halted
-    # will call process method of a proper backend
+    # will call perform method of a proper backend
     def call
-      run_callbacks :after_received do
-        process
-      end
+      process
```

Comment:
If it says that it will call `perform` why is it calling `process`?

Reply:
Wrong method description. ```call``` calls process. ```process``` is a method for processing data that is provided by the engine, so the process is not defined on the controller, but provided by the engine. For inline it will just call ```consume``` BUT for sidekiq it behaves differently.